common/actions.js

```diff
@@ -30,6 +30,21 @@ export const addUserSays = () => {
   }
 }
 
+export const addResponse = () => {
+  return {
+    type: "ADD_RESPONSE",
+    id: intentId++
+  }
+}
+
+export const addConversationComponent = (cType) => {
+  return {
+    type: "ADD_CONVERSATION_COMPONENT",
+    id: (intentId+=10)-10,
```

Comment:
Did you choose the number 10 because you prefer 10 over 9? I did count the numbers of nodes and edges for each conversation component and realized the maximum number is 9.